### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by gjbunyea**
\#108

# Description

Added linting and linting-related workflows. Follows recommended eslint config with several custom rules to enforce **no semi, single quotes, 2 space indent** among others. Also a few react-specific config so jsx is properly linted.

# Change Log
Describe all changes made 

- `eslint.json` : New file - added basic config with several custom rules
- `package.json` : added `lint` and `lint-w` scripts
- `.travis.yml` : added linting to travis process
- `README.md` : added how to lint section, fixed one quote
- `src/...` : all src files are free of linting errors. 

# Manual Test Steps
1. `npm install`
2. `npm run lint` - Notice: no errors
3. Make a style error (change `'` to `"` somewhere, indent too much...)
4. run `./node_modules/.bin/eslint src/` - Notice: your error is reported
5. `npm run lint` - Notice: no errors, Notice: your error has been corrected in the file
6. `npm run lint-w` starts auto-lint and updates when file changed

### CODE REVIEW

**Comment by gjbunyea** [comment]
We won't be able to catch one line breaks with ESLint, it'll have to be checked on PR

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  "env": {
+    "browser": true,
+    "node": false
+  },
+  "extends": [
+    "eslint:recommended"
+  ],
+  "rules": {
+    "brace-style": 1,
+    "comma-dangle": 1,
```
Comment (angushtlam) [inline .eslintrc.json]: It looks like the linter is not supporting comma dangle properly without the full config

"comma-dangle": ["error", "always-multiline"]

https://eslint.org/docs/rules/comma-dangle

File: src/view/components/white_button.js [hunk 2]
```diff
@@ -3,60 +3,60 @@ import {Button} from 'semantic-ui-react';
...
-        this.hover = this.hover.bind(this);
-        this.unhover = this.unhover.bind(this);
-        this.state = {
-            hovering: false,
-        }
-    }
+    this.hover = this.hover.bind(this);
+    this.unhover = this.unhover.bind(this);
+    this.state = {
+      hovering: false
```
Comment (angushtlam) [inline src/view/components/white_button.js]: Dangling comma should be supported

File: src/view/visualizer/sketch/index.js [hunk 3]
```diff
@@ -1,195 +1,195 @@
...
-      case 4: return '#ff96b4'
-      case 5: return '#0096b4'
-      case 6: return '#0069fb'
-      case 7: return '#af96fb'
-      case 8: return '#ff8b6a'
-      case 9: return '#ffbf00'
-      case 10: return '#ff00c3'
-      case 11: return '#0000c3'
-      default: return '#00c9fe'
+    case 0: return '#00c9fe';
```
Comment (angushtlam) [inline src/view/visualizer/sketch/index.js]: Do we want to remove the indent from the switch case?

Comment (gjbunyea) [reply src/view/visualizer/sketch/index.js]: I have no opinion on this - @zsmoore @noahpoz @AlexPerri

**Pull request merged**